src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,145 @@
+package nextstep.subway.domain;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import nextstep.subway.exception.BothStationAlreadyExistsException;
+import nextstep.subway.exception.BothStationNotExistsException;
+import nextstep.subway.exception.DistanceIsEqualOrGreaterException;
+import nextstep.subway.exception.ResourceNotFoundException;
+
+@Embeddable
+public class Sections {
+    @OneToMany(mappedBy = "line", cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+
+    public boolean addSection(final Section addableSection) {
+        checkBothStationExists(addableSection);
+        if (checkSectionIsBetween(addableSection)) {
+            checkDistanceIsEqualOrGreaterThan(addableSection);
+            updateOriginSection(addableSection);
+        }
+        return sections.add(addableSection);
+    }
+
+    private void checkBothStationExists(final Section addableSection) {
+        if (sections.isEmpty()) {
+            return;
+        }
+
+        final long matchedCount = getAllStations().stream()
+                .filter(station ->
+                        station == addableSection.getUpStation() ||
+                                station == addableSection.getDownStation())
+                .count();
+
+        if (matchedCount == 2) {
+            throw new BothStationAlreadyExistsException();
+        }
```

Comment:
```suggestion
      validateDuplicate();
```
같은 함수로 좀 더 명확하게 분리시켜보면 어떨까요?